### PR TITLE
feat:implement front end connect to crud

### PR DESCRIPTION
**Pull request by SalomonKein**

### CODE REVIEW

File: mobile/App.tsx [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
 import {StyleSheet, Text, View} from 'react-native';
+import Header from './src/components/Header';
+import MainStack from './src/navigations/navigations';
+
+export const Context = React.createContext();
+
 export default function App() {
+  const [isUpdate, setIsUpdate] = useState(false)
+
+  function updComponent() {
```
Comment (CodeGeneration-2020) [inline mobile/App.tsx:12]: Whaat is that ?

File: mobile/App.tsx [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
+import { StatusBar } from 'expo-status-bar';
+import React, { useState } from 'react';
 import {StyleSheet, Text, View} from 'react-native';
+import Header from './src/components/Header';
+import MainStack from './src/navigations/navigations';
+
+export const Context = React.createContext();
+
 export default function App() {
+  const [isUpdate, setIsUpdate] = useState(false)
```
Comment (CodeGeneration-2020) [inline mobile/App.tsx:10]: Why do you need this state ?

File: mobile/src/components/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      <View style={styles.textContainer}>
+        <Text style={styles.text}>Todo App</Text>
+      </View>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: "#22b44b",
```
Comment (CodeGeneration-2020) [inline mobile/src/components/Header.tsx:16]: Move to the the constants as we discussed

File: mobile/src/components/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    width: "100%",
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  textContainer: {
+    paddingTop: "10px",
```
Comment (CodeGeneration-2020) [inline mobile/src/components/Header.tsx:23]: Move to the the constants as we discussed

File: mobile/src/components/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    paddingBottom: "10px",
+  },
+  text: {
+    fontWeight: "bold",
+    color: "white",
```
Comment (CodeGeneration-2020) [inline mobile/src/components/Header.tsx:28]: Move to the the constants as we discussed

File: mobile/src/components/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    fontSize: 20,
```
Comment (CodeGeneration-2020) [inline mobile/src/components/Header.tsx:29]: Move to the the constants as we discussed

File: mobile/src/components/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  textContainer: {
+    paddingTop: "10px",
+    paddingBottom: "10px",
+  },
+  text: {
+    fontWeight: "bold",
```
Comment (CodeGeneration-2020) [inline mobile/src/components/Header.tsx:27]: Move to the the constants as we discussed

File: mobile/src/components/TodoItem.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import {Context} from "../../App";
+import {TodoService} from "../services/todo.services";
+
+const TodoItem = ({
+  item,
+  onPress,
+  backgroundColor,
+  textColor,
+}: {
+  item: any;
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:22]: Do not use any

File: mobile/src/components/TodoItem.tsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  onPress: any;
+  backgroundColor: any;
+  textColor: any;
+}) => {
+  const {updComponent} = useContext(Context);
+  const navigation = useNavigation();
+
+  const deleteTodo = () => {
+    updComponent();
+    const deletePromis = new TodoService();
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:32]: async await

File: mobile/src/components/TodoItem.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  item: any;
+  onPress: any;
+  backgroundColor: any;
+  textColor: any;
+}) => {
+  const {updComponent} = useContext(Context);
+  const navigation = useNavigation();
+
+  const deleteTodo = () => {
+    updComponent();
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:31]: What is that ?

File: mobile/src/components/TodoItem.tsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+      <View style={styles.contentContainer}>
+        {!item.isCompleted ? (
+          <Text style={[styles.title, textColor]}>{item.title}</Text>
+        ) : (
+          <Text style={[styles.titleThrow, textColor]}>{item.title}</Text>
+        )}
+        <Text style={styles.content}>{item.description}</Text>
+      </View>
+      <View style={styles.buttonContainer}>
+        <Button title="Update" color="#22b44b" onPress={() => updTodo()} />
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:57]: Move to the the constants as we discussed

File: mobile/src/components/TodoItem.tsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        <Button title="Delete" color="red" onPress={() => deleteTodo()} />
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:58]: Move to the the constants as we discussed

File: mobile/src/components/TodoItem.tsx [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+      </View>
+    </TouchableOpacity>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    marginTop: StatusBar.currentHeight || 0,
+    backgroundColor: "#E6E6FA",
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:68]: Move to the the constants as we discussed

File: mobile/src/components/TodoItem.tsx [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+  },
+  item: {
+    display: "flex",
+    flexDirection: "row",
+    padding: 20,
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:73]: Move to the the constants as we discussed

File: mobile/src/components/TodoItem.tsx [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+    marginVertical: 8,
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:74]: Move to the the constants as we discussed

File: mobile/src/components/TodoItem.tsx [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+    padding: 10,
+    height: 50,
+  },
+  buttonContainer: {
+    display: "flex",
+    justifyContent: "space-between",
+    marginLeft: 10,
+  },
+  button: {
+    margin: "5px",
```
Comment (CodeGeneration-2020) [inline mobile/src/components/TodoItem.tsx:98]: Move to the the constants as we discussed

File: mobile/src/pages/Auth.tsx [hunk 17]
```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import {Formik} from "formik";
+import {StyleSheet, View, Button} from "react-native";
+import * as yup from "yup";
+
+const Registration = ({navigation}: {navigation: any}) => {
+  const validationsSchema = yup.object().shape({
+    email: yup
```
Comment (CodeGeneration-2020) [inline mobile/src/pages/Auth.tsx:8]: move validation to the separate file

File: mobile/src/pages/Auth.tsx [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+          touched,
+          handleChange,
+          handleBlur,
+          isValid,
+          handleSubmit,
+          dirty,
+        }) => (
+          <View style={styles.formContainer}>
+            <p>
+              <label htmlFor={`email`}>Email</label>
```
Comment (CodeGeneration-2020) [inline mobile/src/pages/Auth.tsx:44]: Create Shared input component with label, errr handling and input itself

File: mobile/src/pages/CreateTodo.tsx [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+    navigation.navigate("TodoList");
+  };
+
+  return (
+    <View style={styles.container}>
+      <Formik
+        initialValues={{
+          title: "",
+          description: "",
+          year: "",
```
Comment (CodeGeneration-2020) [inline mobile/src/pages/CreateTodo.tsx:25]: Missed validation

File: mobile/src/pages/CreateTodo.tsx [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+          isCompleted: false,
+          isPrivate: false,
+        }}
+        onSubmit={(values: Values, {setSubmitting}: FormikHelpers<Values>) => {
+          setTimeout(() => {
```
Comment (CodeGeneration-2020) [inline mobile/src/pages/CreateTodo.tsx:30]: Set timeout ? Why ?

Comment (CodeGeneration-2020) [reply mobile/src/pages/CreateTodo.tsx:30]: Use react query

File: mobile/src/pages/UpdateTodo.tsx [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+
+interface Values {
+  title: string;
+  description: string;
+  year: string;
+  isCompleted: boolean;
+  isPrivate: boolean;
+}
+
+const UpdateTodo = ({navigation, route}: {navigation: any; route: any}) => {
```
Comment (CodeGeneration-2020) [inline mobile/src/pages/UpdateTodo.tsx:14]: any

File: mobile/src/services/todo.services.ts [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+  }
+  async createTodo(todo: any) {
+    const todoCreate = await this.post({
+      url: "todos",
+      data: todo,
+      headers: undefined,
+    });
+    return todoCreate;
+  }
+  async updateTodo(todo: any) {
```
Comment (CodeGeneration-2020) [inline mobile/src/services/todo.services.ts:27]: any

File: mobile/src/services/todo.services.ts [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+    });
+    return todo;
+  }
+  async getTodos() {
+    const todos = await this.get({
+      url: "todos",
+    });
+    return todos;
+  }
+  async createTodo(todo: any) {
```
Comment (CodeGeneration-2020) [inline mobile/src/services/todo.services.ts:19]: any

File: mobile/src/services/user.services.ts [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+  constructor() {
+    super();
+  }
+  async getUser(id: string) {
+    const user = await this.get({
+      url: `user/${id}`,
+    });
+    return user;
+  }
+  async createUser(user: any) {
```
Comment (CodeGeneration-2020) [inline mobile/src/services/user.services.ts:13]: any

**Review by CodeGeneration-2020 — CHANGES_REQUESTED** [review]

File: mobile/src/components/Header.tsx (hunk 3 above)
Comment (SalomonKein) [reply mobile/src/components/Header.tsx:16]: I don't understand what the discussion is about, maybe I was not present at this

**Pull request closed without merging**